### ISSUE TITLE
Bug Report: Sky Atmosphere Blinking Black on World Z value changed in Editor.

### ISSUE THREAD

**Issue by KylerSF** [o3de/o3de#12684]
**Describe the bug**
The Sky Atmosphere component blinks black every once in a while when traversing the Z coordinate in the editor.

**Assets required**
Sky Atmosphere Gem

**Steps to reproduce**
1. Add Sky Atmosphere component to an entity
2. Traverse the Z coordinate with E and Q keys.
3. See the sky atmosphere blink black.

**Expected behavior**
There is no blinking in the sky atmosphere.

**Actual behavior**
There is blinking in the sky atmosphere.

**Screenshots/Video**
https://user-images.githubusercontent.com/100494538/196585331-10b58ef3-bfac-4934-acec-a5a3dbdd2851.mp4

**Found in Branch**
Development [ce8bfb7]

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version: 11

**Comment by AMZN-alexpete** [comment]
Hi @ParasiteSt I can not reproduce this error in latest development branch.  Can you share your level so I can verify I'm using the same atmosphere settings?  It's also possible this issue was recently fixed.

**Comment by ParasiteSt** [comment]
Hey @AMZN-alexpete sorry for the late reply I was trying to dig into a few problems, this being one of them. I updated to the newest dev thinking the new change you added may have solved this issue, but it did not. The issue is fully based on if the rotation is set to -90 on the X axis for the component holding the sky atmosphere component. I did not know this because I have the entity set to using a light/sun source as rotation, and in editor that rotation is set to -90 for a noon like atmosphere.

**Comment by AMZN-alexpete** [comment]
Fix for this is coming in the linked PR #13815 - there was a nan introduced in this line when sunZenithCosAngle was close to 1.0
```
SunDir = normalize(float3(sqrt(1.0 - sunZenithCosAngle * sunZenithCosAngle), 0.0, sunZenithCosAngle));
```
